typescript/infra/scripts/migrate-funds-from-aws-keys.ts

```diff
@@ -0,0 +1,164 @@
+            multiProvider,
+            fromKey,
+            toKey,
+          );
+        }
+      }
+    }
+  }
+}
+
+async function transfer(
+  chain: ChainName,
+  agentConfig: AgentContextConfig,
+  igp: HyperlaneIgp,
+  multiProvider: MultiProvider,
+  fromKey: CloudAgentKey,
+  toKey: CloudAgentKey,
+) {
+  if (chainMetadata[chain].protocol != ProtocolType.Ethereum) return;
+  const logCtx: any = {
+    chain,
+    from: fromKey.identifier,
+    fromKey: fromKey.address,
+    to: toKey.identifier,
+    toKey: toKey.address,
+  };
+  console.log('Processing key', logCtx);
+
+  const igpContract = igp.getContracts(chain).interchainGasPaymaster;
+
+  const igpBalance = await multiProvider
+    .getProvider(chain)
+    .getBalance(igpContract.address);
+  const claimIgpFundsTx = await igpContract.populateTransaction.claim();
+  const gasToClaimIgpFunds = await multiProvider.estimateGas(
+    chain,
+    claimIgpFundsTx,
+  );
+  let gasPrice = await multiProvider.getProvider(chain).getGasPrice();
+  const costToClaimIgpFunds = gasPrice.mul(gasToClaimIgpFunds);
+  if (igpBalance.gt(costToClaimIgpFunds)) {
+    // only claim if the cost to do so is less than the balance we are claiming
+    // await multiProvider.sendTransaction(chain, claimIgpFundsTx);
+    console.log('Claimed IGP funds', {
+      ...logCtx,
+      igpAddress: igpContract.address,
+      igpBalance: formatEther(igpBalance),
+      cost: formatEther(costToClaimIgpFunds),
+      tx: claimIgpFundsTx,
+    });
+    // update gas price since we might have waited a while
+    gasPrice = await multiProvider.getProvider(chain).getGasPrice();
+  } else {
+    console.log('IGP balance too low to claim', {
+      ...logCtx,
+      igpAddress: igpContract.address,
+      igpBalance: formatEther(igpBalance),
+      cost: formatEther(costToClaimIgpFunds),
+    });
+  }
+
+  const currentBalance = await multiProvider
+    .getProvider(chain)
+    .getBalance(fromKey.address);
+  const transferTx: PopulatedTransaction = {
+    to: toKey.address,
+    value: BigNumber.from(0),
+  };
+  const gasToTransfer = await multiProvider.estimateGas(chain, transferTx);
+  const costToTransfer = gasToTransfer.mul(gasPrice);
+  if (costToTransfer.gt(currentBalance)) {
+    console.log('Not enough funds to transfer', {
+      ...logCtx,
+      balance: formatEther(currentBalance),
+    });
+    return;
+  }
+  transferTx.value = currentBalance.sub(costToTransfer);
+  transferTx.gasLimit = gasToTransfer;
+  // await multiProvider.sendTransaction(chain, transferTx);
+  console.log('Transferred funds', {
+    ...logCtx,
+    originalBalance: formatEther(currentBalance),
+    finalBalance: formatEther(
+      await multiProvider.getProvider(chain).getBalance(fromKey.address),
+    ),
+    destinationBalance: formatEther(
+      await multiProvider.getProvider(chain).getBalance(toKey.address),
+    ),
+    cost: formatEther(costToTransfer),
+    transferTx,
+  });
+  // await fromKey.delete();
```

Comment:
I assume you'll send the transferTx here?

May wanna consider running things on a cheap chain with `transferTx.value = 1` first so you can confirm everything looks right. at least my paranoid self would 😄